packages/devtools/src/sourceMap/loadOriginalSourceLoc.ts

```diff
@@ -0,0 +1,178 @@
+  }
+
+  try {
+    return extractAndLoadSourceMapJSON(sourceLoc);
+  } catch (error) {
+    console.warn(`[Griffel devtools] unable to load source map for ${sourceUrlWithLoc}`);
+    console.warn(error);
+    return sourceLoc;
+  }
+}
+
+// inspired by https://github.com/facebook/react/blob/b66936ece7d9ad41a33e077933c9af0bda8bff87/packages/react-devtools-shared/src/hooks/parseHookNames/loadSourceAndMetadata.js#L135
+async function extractAndLoadSourceMapJSON(sourceLoc: MappedPosition): Promise<MappedPosition> {
+  const { source, line, column } = sourceLoc;
+
+  const runtimeSourceCode = await fetchFiles(source);
+
+  const externalSourceMapURLs = [];
+  let sourceMapNum = 0;
+  const sourceMapRegex = / ?sourceMappingURL=([^\s'"]+)/gm;
+
+  let sourceMappingURLMatch = sourceMapRegex.exec(runtimeSourceCode);
+  if (!sourceMappingURLMatch) {
+    throw new Error(`[Griffel devtools] extractAndLoadSourceMapJSON() no source map found in ${source}`);
+  }
+
+  while (sourceMappingURLMatch) {
+    sourceMapNum++;
+    const sourceMappingURL = sourceMappingURLMatch[1];
+
+    const hasInlineSourceMap = sourceMappingURL.indexOf('base64,') >= 0;
+    if (hasInlineSourceMap) {
+      try {
+        // Web apps like Code Sandbox embed multiple inline source maps.
+        // In this case, we need to loop through and find the right one.
+        const trimmed = sourceMappingURL.match(/base64,([a-zA-Z0-9+/=]+)/)?.[1] ?? '';
+        const decoded = decodeBase64String(trimmed);
+        const sourceMapJSON = JSON.parse(decoded);
+
+        // TODO optionally turn on debug message
+        // if (__DEBUG__) {
+        //   console.groupCollapsed('[Griffel devtools] extractAndLoadSourceMapJSON() Inline source map');
+        //   console.log(sourceMapJSON);
+        //   console.groupEnd();
+        // }
+
+        if (sourceMapIncludesSource(sourceMapJSON, getFilePath(source))) {
+          // store computed sourceMapJSON
+          if (sourceMapNum > 1) {
+            sourceMapJSONs.set(`${source}:${line}:${column}`, sourceMapJSON);
+          } else {
+            if (sourceMapRegex.exec(runtimeSourceCode) === null) {
+              sourceMapJSONs.set(source, sourceMapJSON);
+            } else {
+              sourceMapJSONs.set(`${source}:${line}:${column}`, sourceMapJSON);
+            }
+          }
+
+          return getOriginalPosition(sourceMapJSON, sourceLoc);
+        }
+      } catch (error) {
+        // We've likely encountered a string in the source code that looks like a source map but isn't.
+        // Maybe the source code contains a "sourceMappingURL" comment or something similar.
+        // In either case, let's skip this and keep looking.
+      }
+    } else {
+      externalSourceMapURLs.push(sourceMappingURL);
+    }
+
+    sourceMappingURLMatch = sourceMapRegex.exec(runtimeSourceCode);
+  }
+
+  // No inline sourcemap found. Check if there's external source map
+  let sourceMappingURL = externalSourceMapURLs.pop();
```

Comment:
why only use the latest external source map and skip all others ?

Reply:
This is also stoled from react-devtools https://github.com/facebook/react/blob/b66936ece7d9ad41a33e077933c9af0bda8bff87/packages/react-devtools-shared/src/hooks/parseHookNames/loadSourceAndMetadata.js#L239

They have a comment there
```tsx
// Files with external source maps should only have a single source map.
// More than one result might indicate an edge case,
// like a string in the source code that matched our "sourceMappingURL" regex.
// We should just skip over cases like this.
```
I will added the same comment too

Reply:
Added in L113